astropy/modeling/parameters.py

```diff
@@ -608,62 +739,36 @@ def __nonzero__(self):
 
     __bool__ = __nonzero__
 
-    def __add__(self, val):
-        return self.value + val
-
-    def __radd__(self, val):
-        return val + self.value
-
-    def __sub__(self, val):
-        return self.value - val
-
-    def __rsub__(self, val):
-        return val - self.value
-
-    def __mul__(self, val):
-        return self.value * val
-
-    def __rmul__(self, val):
-        return val * self.value
-
-    def __pow__(self, val):
-        return self.value ** val
-
-    def __rpow__(self, val):
-        return val ** self.value
-
-    def __div__(self, val):
-        return self.value / val
-
-    def __rdiv__(self, val):
-        return val / self.value
-
-    def __truediv__(self, val):
-        return self.value / val
-
-    def __rtruediv__(self, val):
-        return val / self.value
-
-    def __eq__(self, val):
-        return (np.asarray(self) == np.asarray(val)).all()
-
-    def __ne__(self, val):
-        return not (np.asarray(self) == np.asarray(val)).all()
-
-    def __lt__(self, val):
-        return (np.asarray(self) < np.asarray(val)).all()
-
-    def __gt__(self, val):
-        return (np.asarray(self) > np.asarray(val)).all()
-
-    def __le__(self, val):
-        return (np.asarray(self) <= np.asarray(val)).all()
-
-    def __ge__(self, val):
-        return (np.asarray(self) >= np.asarray(val)).all()
+    __add__ = _binary_arithmetic_operation(operator.add)
+    __radd__ = _binary_arithmetic_operation(operator.add, True)
+    __sub__ = _binary_arithmetic_operation(operator.sub)
+    __rsub__ = _binary_arithmetic_operation(operator.sub, True)
+    __mul__ = _binary_arithmetic_operation(operator.mul)
+    __rmul__ = _binary_arithmetic_operation(operator.mul, True)
+    __pow__ = _binary_arithmetic_operation(operator.pow)
+    __rpow__ = _binary_arithmetic_operation(operator.pow, True)
+    __div__ = _binary_arithmetic_operation(operator.truediv)
+    __rdiv__ = _binary_arithmetic_operation(operator.truediv, True)
+    __truediv__ = _binary_arithmetic_operation(operator.truediv)
+    __rtruediv__ = _binary_arithmetic_operation(operator.truediv, True)
+    __eq__ = _binary_comparison_operation(operator.eq)
+    __ne__ = _binary_comparison_operation(operator.ne)
+    __lt__ = _binary_comparison_operation(operator.lt)
+    __gt__ = _binary_comparison_operation(operator.gt)
+    __le__ = _binary_comparison_operation(operator.le)
+    __ge__ = _binary_comparison_operation(operator.ge)
+    __neg__ = _unary_arithmetic_operation(operator.neg)
+    __abs__ = _unary_arithmetic_operation(operator.abs)
+
+
+def param_repr_oneline(param):
+    """
+    Like array_repr_oneline but works on `Parameter` objects and supports
+    rendering parameters with units like quantities.
+    """
 
-    def __neg__(self):
-        return -self.value
+    out = array_repr_oneline(param.value)
+    if param.unit is not None:
+        out = '{0} {1!s}'.format(out, param.unit)
```

Comment:
This will look ugly for dimensionless units. How about: `out = Quantity.__str__(param)`
or, copying `Quantity._unitstr`,
```
    unitstr = str(param.unit)
    if unitstr:
        unitstr = ' ' + unitstr
    out = {0}{1}'.format(out, unitstr)
```